lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriterOptions.java

```diff
@@ -35,8 +34,8 @@ public static ParquetWriterOptions.Builder builder()
 
     private ParquetWriterOptions(DataSize maxBlockSize, DataSize maxPageSize, int batchSize)
     {
-        this.maxRowGroupSize = toIntExact(maxBlockSize.toBytes());
-        this.maxPageSize = toIntExact(maxPageSize.toBytes());
+        this.maxRowGroupSize = Ints.saturatedCast(maxBlockSize.toBytes());
+        this.maxPageSize = Ints.saturatedCast(maxPageSize.toBytes());
```

Comment:
I don't think we want users to configure large page sizes beyond a few MBs. We could add the below annotations in ParquetWriterConfig for `parquet.writer.page-size`
```
@MinDataSize("16kB")
@MaxDataSize("8MB")
```

Reply:
Makes sense. I changed this based on `AbstractTestParquetPageSkipping` setting `parquet_writer_block_size=100GB`.